Inseok_45/app.js

```diff
@@ -44,12 +41,70 @@ app.post('/users/signup', async (req, res) => {
             age,
             user_name,
             password
-        ) VALUES ( ?, ?, ?, ?, ?, ?, ?);
-        `, [firstName, lastName, email, phoneNumber, age, userName, password]
+        ) VALUES ( ?, ?, ?, ?, ?, ?, ?)`, [firstName, lastName, email, phoneNumber, age, userName, password]
+    );
+
+    return res.status(201).json({ message: "sign-up complete" });
+});
+
+app.post('/users/post', async (req, res) => {
+    const { userId, postImage, postParagraph} = req.body
+
+    await dataSource.query(
+        `INSERT INTO posts(
+            user_id,
+            post_image,
+            post_paragraph
+        ) VALUES ( ?, ?, ?)`, [userId, postImage, postParagraph]
+    );
+
+    return res.status(201).json({ message: "post created!" });
+});
+
+app.get('/users/post/view', async (req, res) => {
+    const viewPost = await dataSource.query(
+        `SELECT
+          users.id as userId,
+          posts.id as postingId,
+          posts.post_image as postingImageUrl,
+          posts.post_paragraph as postingContent
+        FROM users
+        INNER JOIN posts ON posts.user_id = users.id`
     );
-    res.status(201).json({ message: "sign-up complete" });
+
+    res.status(200).json({ data: viewPost });
+});
+
+app.get('/users/:userId/post/view', async (req, res, next) => {
+
+    var userId = req.params.userId;
+
+    const [userPost] = await dataSource.query(
+        `SELECT
+          users.id AS userId,
+          (SELECT 
+            JSON_ARRAYAGG(
+            JSON_OBJECT(
+                "postingId", posts.id,
+                "postImageUrl", posts.post_image,
+                "postContent", posts.post_paragraph
+            ))
+        FROM posts 
+        WHERE posts.user_id = users.id && users.id = ?
```

Comment:
&&을 이용한 and 연산은 mysql에서만 가능한 문법입니다. 따라서 일반적으로 쓰이는 AND로 수정해 주시면 좋을 것 같습니다.
```suggestion
        WHERE posts.user_id = users.id AND users.id = ?
```

Reply:
수정하였습니다!